### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by ZoeMitchelll**
integrating authentication library into my portfolio

### CODE REVIEW

File: portfolio/src/main/webapp/index.html [hunk 1]
```diff
@@ -49,22 +49,27 @@
                 <p>Please feel free to drop some questions, comments, and concerns if you have any:</p>
                 <form action="/my-CQCs" method="POST">
                     <input type="text" name="email-input" placeholder="Your Email"/>
-                    <br/><br/>
+                    <br></br>
```
Comment (lee12jin) [inline portfolio/src/main/webapp/index.html]: For br statements you typically can. just leave it as 

`<br>`

without closing tags or any slashes! It'll still work.

File: portfolio/src/main/webapp/index.html [hunk 2]
```diff
@@ -49,22 +49,27 @@
...
 
                     <select name="type-input">
                         <option value="question">Question</option>
                         <option value="comment">Comment</option>
                         <option value="concern">Concern</option>
                     </select>
-                    <br/><br/>
+                    <br></br>
```
Comment (lee12jin) [inline portfolio/src/main/webapp/index.html]: Same comment here.

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -42,4 +42,18 @@ async function getRandomQuoteUsingAsyncAwait() {
   const response = await fetch('/data');
   const quote = await response.text();
```
Comment (lee12jin) [inline portfolio/src/main/webapp/script.js]: I must have missed this but this doesn't need an `await`.

`const quote = response.text();`

await is just used for code that can be executed (asynchronously). When you reference the variable (in this case, response) in another line, the code execution stops and waits until response is defined before assigning a value to quote.

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -42,4 +42,18 @@ async function getRandomQuoteUsingAsyncAwait() {
...
   document.getElementById('quote-container').innerText = quote;
+}
+
+async function loadTasks() {
+    let list = await fetch('/list-tasks').then(response => response.json());
```
Comment (lee12jin) [inline portfolio/src/main/webapp/script.js]: I don't think it's typical to mix await with promises like this. Stick to one or the other (i.e.):

```
let response = await fetch('/list-tasks')
let list = response.json()
```

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -42,4 +42,18 @@ async function getRandomQuoteUsingAsyncAwait() {
...
+    for(c in list){ 
+        let reviewsElem = document.getElementById("reviews");
+        let commentElem = document.createElement("DIV");
+        commentElem.appendChild(document.createTextNode(list[c].type+" "+list[c].description));
+        reviewsElem.appendChild(commentElem);
+    }
+}
+async function loginStatus() {
+    let status = await fetch('/status').then(response => response.json());
```
Comment (lee12jin) [inline portfolio/src/main/webapp/script.js:57]: Same comment as above :)

**Review by lee12jin — CHANGES_REQUESTED** [review]

File: portfolio/src/main/webapp/script.js (hunk 3 above)
Comment (lee12jin) [reply portfolio/src/main/webapp/script.js]: Actually nevermind, I retract this. I think I see some documentation saying you should keep await on dependent variables. Ignore what I said here.

File: portfolio/src/main/webapp/script.js (hunk 4 above)
Comment (lee12jin) [reply portfolio/src/main/webapp/script.js]: `let list = await response.json()` I think?

File: portfolio/src/main/webapp/index.html [hunk 6]
```diff
@@ -49,14 +49,14 @@
...
```
Comment (lee12jin) [inline portfolio/src/main/webapp/index.html]: Same comment (\<br>) No need to close the tag here.

File: portfolio/src/main/webapp/index.html [hunk 7]
```diff
@@ -49,14 +49,14 @@
                 <p>Please feel free to drop some questions, comments, and concerns if you have any:</p>
                 <form action="/my-CQCs" method="POST">
                     <input type="text" name="email-input" placeholder="Your Email"/>
-                    <br/><br/>
+                    <br></br>
```
Comment (lee12jin) [inline portfolio/src/main/webapp/index.html]: Same comment (\<br>) No need to close the tag here.

File: portfolio/src/main/webapp/script.js (hunk 5 above)
Comment (lee12jin) [reply portfolio/src/main/webapp/script.js:57]: Still seems to be using await and promise.

```
let response = await fetch('/status');
let status = await response.json();
```

**Review by lee12jin — CHANGES_REQUESTED** [review]

File: portfolio/src/main/webapp/script.js (hunk 5 above)
Comment (ZoeMitchelll) [reply portfolio/src/main/webapp/script.js:57]: If I merge commentAuth (which solved this problem) into this branch after my other pull request is closed, would that fix this problem or would there be a conflict in the merge?

Comment (lee12jin) [reply portfolio/src/main/webapp/script.js:57]: I think if you addressed this comment in the commentAuth, you can just make a comment here saying "I did this in the other PR", and not have to merge different branches together.

Ideally, the simplest workflow is to have individual branches only merging in with master, and not with each other.

**Review by lee12jin — APPROVED** [review]
LGTM because it seems the only comment here was fixed in your other PR

File: portfolio/src/main/webapp/script.js (hunk 5 above)
Comment (lee12jin) [reply portfolio/src/main/webapp/script.js:57]: In the future though, it's best you address comments in the same PR/branch. Then for your other PR/branch, Jasper or I will make the effort not to make the same comment again

**Review by jasperlee-google — APPROVED** [review]

**Pull request merged**